examples/atmega2560p/analog/src/main.rs

```diff
@@ -0,0 +1,40 @@
+#![no_std]
+#![no_main]
+#![deny(warnings)]
+
+/// Crates to be used.
+use rustduino::atmega2560p::hal::analog::map_from1023_to255;
+use rustduino::atmega2560p::hal::pin::Pins;
+use rustduino::atmega2560p::hal::watchdog::WatchDog;
+
+#[no_mangle]
+pub fn main() {
+    // Disable watchdog
+    let watchdog = unsafe { WatchDog::new() };
+    watchdog.disable();
+
+    // Creates a array object consisting of all the pins.
+    let mut pins = Pins::new();
+
+    // Infinite loop for read and write continuously through the I/O pins.
+    loop {
+        // Take input into the zeroth analog pin.
+        let a: u32 = pins.analog[0].read();
+
+        // Make the input value ready to be sent through a digital pin.
+        let b: u8 = map_from1023_to255(a);
```

Comment:
Create a map function in `math` :
```c
long map(long x, long in_min, long in_max, long out_min, long out_max) {
  return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
}
```

Reply:
I will adjust this in #55 Random Number Generator issue. As this issue was meant to be done before #55 this branch does not contain math.rs file and adding it would cause problems as this also doesn't consists MPU6050 sensor.

Reply:
Math.rs is not a file, we will have a module math inside the crate. So you can go ahead and create map.rs inside math.

Reply:
okay added